### PR TITLE
Update axios [MAILPOET-6191]

### PR DESCRIPTION
**Pull request by costasovo**
## Description

This PR addresses the Dependabot's report https://github.com/mailpoet/mailpoet/security/dependabot/107
The axios is used by `@wordpress/scripts` 

## Code review notes

- I updated the required pnpm version (`corepack up`). It looks like some of us have been already using a newer version than the one set in package.json. The newer version writes the version for overridden packages directly to the lock file. So, running even a little update in the older version was causing many changes in the lockfile ([example - this was actually from older to newer](https://github.com/mailpoet/mailpoet/pull/5716/files#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bb)). 
- I updated axios
- I ran `pnpm dedupe` after the update. The `pnpm dedupe` updated a couple of minor versions, and Typescript
- I downgraded Typescript back to the current version because the new version requires more updates and configuration changes in eslint. See more info in commit message.

## QA notes

We can skip QA

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-6191]

## After-merge notes

Ping #mailpoetdev 

> Fyi, we updated the required pnpm version for plugins to 8.15.9. Note that it is recommended to use Corepack to manage the version automatically (run `corepack enable`). 
## Tasks

- [x] 🚫  I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] 🚫  I added sufficient test coverage
- [x] 🚫  I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-6191]: https://mailpoet.atlassian.net/browse/MAILPOET-6191?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/segments/dynamic/types.ts [hunk 1]
```diff
@@ -183,6 +183,7 @@ export interface EmailFormItem extends FormItem {
   opens?: string;
   emails?: string;
   clicks?: string;
+  timeframe?: Timeframe;
```
Comment (JanJakes) [inline mailpoet/assets/js/src/segments/dynamic/types.ts:186]: Why was this change needed when we seem to keep the same TS version?

**Review by JanJakes — APPROVED** [review]
LGTM, thanks!

File: mailpoet/assets/js/src/segments/dynamic/types.ts (hunk 1 above)
Comment (costasovo) [reply mailpoet/assets/js/src/segments/dynamic/types.ts:186]: I added it after the update because it was causing error with the updated TS version. I found that this prop was indeed missing in the definition but with the current TS version there is no error. I didn't revert the change as this is an improvement. So it was actually not needed it and I kept it kind of for the future. Sorry for the confusion.
